backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts

```diff
@@ -38,4 +46,159 @@ export class UserCheckMiddleware {
       return next(error);
     }
   }
+
+  /**
+   * Handles potential ephemeral session groups provided in the request.
+   *
+   * This method processes different scenarios based on the combination of
+   * `groupsForSession` and `includeStoredUserGroups` parameters:
+   *
+   * Note: provided session groups are never persisted
+   *
+   * **Scenario 1: Session-only groups (Ephemeral group user request)**
+   * - `groupsForSession`: provided
+   * - `includeStoredUserGroups`: explicitly `false`
+   * - Behavior: Uses ONLY the provided session groups, ignoring any stored user groups
+   * - Use case: Contexts where complete group information is preferred to be provided at runtime,
+   *    rather reading from stored user groups.
+   *
+   * **Scenario 2: Merged groups (Merged stored/ephemeral groups request mode)**
+   * - `groupsForSession`: provided
+   * - `includeStoredUserGroups`: `true`
+   * - Behavior: Merges session groups with stored user groups, if they don't already exist
+   * - Use case: Adding temporary context while preserving existing user groups
```

Comment:
'while preserving...' seems like you're just not overwriting, which none of these do. Maybe, 'in addition to...' would be clearer?